chapters/classes.tex

```diff
@@ -217,17 +217,27 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
-record and connector components -- see also record specialized class, \cref{specialized-classes}.
+record, operator record, and connector components -- see also record specialized class, \cref{specialized-classes}.
 
 An exception is \lstinline!input! for components whose type is of the special class function type (these can only be used for function formal parameters and has special semantics, see \cref{functional-input-arguments-to-functions}), and the \lstinline!input! prefix is not applied to the elements of the component and is allowed even if the elements have \lstinline!input! or \lstinline!output! prefix.
 
 In addition, instances of classes extending from \lstinline!ExternalObject! may have type prefixes \lstinline!parameter! and \lstinline!constant!, and in functions also type prefixes
 \lstinline!input! and \lstinline!output!, see \cref{external-objects}.
 
-The type prefixes \lstinline!flow!, \lstinline!stream!, \lstinline!input! and \lstinline!output! of a structured component (except as described above) are also applied to the elements of the component (this is done after verifying that the type prefixes occurring on elements of the component are correct; e.g.\ the \lstinline!flow! prefix can be used on a record component and all the record elements will generate zero-sum equations, even if elements of a record shall not be declared with the \lstinline!flow! prefix).  When any of the type prefixes \lstinline!flow!, \lstinline!stream!, \lstinline!input! and \lstinline!output! are applied for a structured component, no element of the component may have any of these type prefixes.  The corresponding rules for the type prefixes \lstinline!discrete!, \lstinline!parameter! and \lstinline!constant! are described in \cref{variability-of-structured-entities} for structured components.
+Variables declared with the \lstinline!stream! type prefix shall be a subtype of \lstinline!Real!.
+Variables declared with the \lstinline!flow! type prefix shall be a subtype of \lstinline!Real! or an operator record definining certain operators, see \cref{generation-of-connection-equations}.
+This is further restricted in \cref{definition-of-stream-connectors}.
+
+The type prefixes \lstinline!input! and \lstinline!output! of a structured component (except as described above) are also applied to the elements of the component (this is done after verifying that the type prefixes occurring on elements of the component are correct).
+When any of the type prefixes \lstinline!input! and \lstinline!output! are applied for a structured component, no element of the component may have any of these type prefixes, nor can they have \lstinline!flow! or \lstinline!stream! prefixes.
+The corresponding rules for the type prefixes \lstinline!discrete!, \lstinline!parameter! and \lstinline!constant! are described in \cref{variability-of-structured-entities} for structured components.
+\begin{nonnormative}
```

Comment:
Empty lines on both sides of `nonnormative` tend to make the source more readable:
```suggestion

\begin{nonnormative}
```